response.go

```diff
@@ -1,85 +1,98 @@
-	Reply(text string, options ...ReplyOption) error
-	ReportError(err error, options ...ReportErrorOption)
+// newWriterReplierResponse creates a new response structure
+func newWriterReplierResponse(poster Writer, replier Replier) WriterReplierResponse {
+	return &writerReplierResponse{poster: poster, replier: replier}
 }
 
-// NewResponse creates a new response structure
-func NewResponse(botCtx BotContext) ResponseWriter {
-	return &response{botCtx: botCtx}
+type writerReplierResponse struct {
+	poster  Writer
+	replier Replier
 }
 
-type response struct {
-	botCtx BotContext
+// Reply send a message to the current channel
+func (r *writerReplierResponse) Reply(message string, options ...ReplyOption) (string, error) {
+	return r.replier.Reply(message, options...)
 }
 
-// ReportError sends back a formatted error message to the channel where we received the event from
-func (r *response) ReportError(err error, options ...ReportErrorOption) {
-	defaults := NewReportErrorDefaults(options...)
+// ReplyError send an error to the current channel
+func (r *writerReplierResponse) ReplyError(err error, options ...ReplyOption) (string, error) {
+	return r.replier.ReplyError(err, options...)
+}
 
-	apiClient := r.botCtx.APIClient()
-	event := r.botCtx.Event()
+// ReplyBlocks send blocks to the current channel
+func (r *writerReplierResponse) ReplyBlocks(blocks []slack.Block, options ...ReplyOption) (string, error) {
+	return r.replier.ReplyBlocks(blocks, options...)
+}
 
-	opts := []slack.MsgOption{
-		slack.MsgOptionText(fmt.Sprintf(errorFormat, err.Error()), false),
-	}
+// Post send a message to a channel
+func (r *writerReplierResponse) Post(channel string, message string, options ...PostOption) (string, error) {
+	return r.poster.Post(channel, message, options...)
+}
 
-	if defaults.ThreadResponse {
-		opts = append(opts, slack.MsgOptionTS(event.TimeStamp))
-	}
+// PostError send an error to a channel
+func (r *writerReplierResponse) PostError(channel string, err error, options ...PostOption) (string, error) {
+	return r.poster.PostError(channel, err, options...)
+}
 
-	_, _, err = apiClient.PostMessage(event.ChannelID, opts...)
-	if err != nil {
-		log.Printf("failed posting message: %v\n", err)
-	}
+// PostBlocks send blocks to a channel
+func (r *writerReplierResponse) PostBlocks(channel string, blocks []slack.Block, options ...PostOption) (string, error) {
+	return r.poster.PostBlocks(channel, blocks, options...)
 }
 
-// Reply send a message to the current channel
-func (r *response) Reply(message string, options ...ReplyOption) error {
-	ev := r.botCtx.Event()
-	if ev == nil {
-		return fmt.Errorf("unable to get message event details")
-	}
-	return r.Post(ev.ChannelID, message, options...)
+// Delete deletes a message in a channel
+func (r *writerReplierResponse) Delete(channel string, messageTimestamp string) (string, error) {
+	return r.poster.Delete(channel, messageTimestamp)
+}
+
+// WriterResponse interface is used to respond to an event
```

Comment:
Not sure I like the name of this. Maybe `PostDeleter` (like `io.ReadWriter`) or a less confusing `PosterDeleter`?

Reply:
Renamed it